62.md

```diff
@@ -0,0 +1,106 @@
+NIP-62
+======
+
+Curated Publication
+----------
+
+`draft` `optional`
+
+This NIP defines the minimum specification for curated publication - ordered, optionally-hierarchical assemblies of nostr events. Collections provide a standard way to organize and present related content, similar to how books organize chapters or journals organize articles.
+
+## Event Kinds
+
+This NIP defines two primary event kinds:
+
+- `30040`: Publication Index
+- `30041`: Publication Content
+
+## kind 30040: Publication Index
+
+A publication index defines the structure and metadata of a collection. It serves as a table of contents that references the actual content sections.
+
+### Requirements
+
+- The `content` field MUST be empty
+- MUST include a `title` tag containing the full title of the collection
+- MUST be uniquely identifiable by the combination of:
+  - `d` tag
+  - `pubkey`
+  - `kind`
+- MUST include `e` tags listing the events in desired display order
+  - Referenced events SHOULD be kind 30041 sections or nested kind 30040 indices
+  - Additional event kinds MAY be supported
+- For derivative works:
+  - MUST include a `p` tag identifying the original author
+  - MUST include an `E` tag referencing the original event immediately after the `p` tag
+
+### Example Index
+
+```json
+{
+    "id": "<event_id>",
+    "pubkey": "<event_originator_pubkey>",
+    "created_at": 1725087283,
+    "kind": 30040,
+    "tags": [
+        ["d", "aesop's-fables-by-aesop"],
+        ["title", "Aesop's Fables"],
+        ["author", "Aesop"],
+        ["i", "isbn:9780765382030"],
+        ["t", "fables"],
+        ["t", "classical"],
+        ["t", "literature"],
+        ["published_on", "2003-05-03"],
+        ["published_by", "public domain"],
+        ["image", "https://imageserver.com/piclink.jpg"],
+        ["summary", "Collection of selected fables from the ancient Greek philosopher, known as Aesop."],
+        ["e", "<id>", "<relay_url>", "<pubkey>"],
```

Comment:
Being forced to change the `kind:30040` every time any `kind:30041` might be a bit annoying; I would use the `a` tag here or at least have it as an option (I,.e. if you were referring to someone else's event you might want to anchor it to a very specific version in which case an `e` tag would be useful)

Reply:
Yes, we should maybe note that a tags can be used instead of e tags. @limina1

Reply:
yes, let's move this to a tags. 

Should it also have a description to render without having to load the attached event?

Reply:
I don't know. Sort of turns it into a list, rather than a publication. I mean, anyone can add a `description` tag to 30041s, of course, and use it that way, but that's a specific implementation out of scope of the NIP.

Reply:
We've added `a` tags as an alternative, but we need to keep `e` tags for research papers and other technical documents.

Reply:
ohh interesting.. what happens if an implementation adds both `a` and `e`? Adding both is quite common when referencing replaceable events. It gives the receiving client an opportunity to update to the latest while also letting the user know that the event has changed.

Reply:
Great explanation - Institutionally, some use cases work well as frozen snapshots. Gives pressure and credibility for things being correct the first time, with errata being a formal but resistive way to update that snapshot

Reply:
Also... keep in mind that most relays will delete past versions of replaceables. If the index is using `e` it is possible and the id is not available anymore, there is not way to find out what is the next version. The user will have to remember what was there, figure out what d-tag that item was using and look for it to get the next id.

Reply:
How about `["a", "kind:pubkey:dtag", "relay hint", "event id"]`? 

In that way, we avoid duplicating `a`'s and `e`'s in the index or clients having to figure out which `e` belongs to which `a`. Clients can highlight that versions have changed based on the "security" required by the document. If they need to update, then all the information is there to find the new event id.

Reply:
I think that's quite an elegant solution

Reply:
Do we need to offer a spec for the desired security model? 

We could have a tag on the index `["auto-update", "yes|ask|no"]`

Reply:
@SilberWitch  @buttercat1791 thoughts? gives users and publishers fine grained control over how to update the events

Reply:
The security model tag could be changed, later, after all, but only if the note contains all of the necessary data.

Reply:
> - MUST include `a` tags listing the events in desired display order.
>   - Format: `["a", "kind:pubkey:dtag", "relay hint", "event id"]`
>   - Optional event ID field enables version tracking while maintaining updateable references
>   - Referenced events SHOULD be kind 30041 sections or nested kind 30040 indices
>   - Additional event kinds MAY be supported
> - MUST include a `auto-update` tag specifying update behavior:
>   - Format: `["auto-update", "<yes|ask|no>"]`
>   - Controls whether clients should automatically update to newer versions

How does this description hold up?

Reply:
"MUST include a auto-update tag"

should be "MUST include an auto-update tag"

Reply:
I agree with y'all, the solution proposed is quite elegant.

The one thing I'll add to this discussion is that we shouldn't remove `e` tags from the specification entirely.  We can reference 30041 and other replaceable event kinds with `a` tags, but some use cases of the `30040` kind may require referencing non-replaceable events, in which case `e` will suffice and we don't need to worry about content changing.

Reply:
I don't like this idea of using 30040 for other kinds. Because then clients that implement this spec must also implement every other kind that gets attached to it. Which is terrible for interoperability. 

Do we have real uses of other kinds in this? If so, let's work through each of them and put the guidance in this nip.

Reply:
I have a wishlist item. I would like to be able to use kind 1063, so that one could put together a journal article that has text sections and media sections. I am aware that images can be added directly into content, but this works somewhat like stock images or custom-made graphics that can be attributed directly. If this is out of scope, I understand.

Reply:
I don't think it is out of scope. Anything text, tables, code, etc., can be in a 30041 (or migrated to, or embedded in, one), but something like an image is sort of extra.

Numerous artists have complained about their images being altered/skewed/cropped by clients, so we should maybe treat images with more care. Identifying something explicitly as media would also ease formatting options like text-wrapping and floating.

Reply:
> I don't like this idea of using 30040 for other kinds. Because then clients that implement this spec must also implement every other kind that gets attached to it. Which is terrible for interoperability.
>
> Do we have real uses of other kinds in this? If so, let's work through each of them and put the guidance in this nip.

Perhaps this NIP will need to be updated over time to provide helpful guidance for integrating with various event kinds.

Right off the bat, though, an obvious case for `30040` events referencing a kind other than `30041` would be to create curated publications that republish long-form articles.  A user might use the NIP-62 hierarchical curation structure to set up a magazine publication that includes a mix or original articles written using kind `30041`, and republished, older articles written in `30023`.

At some point, the client developer can determine how far to extend the supported features beyond the basic `30040` and `30041` pairing specified in this NIP.